feature-libs/product/configurators/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.spec.ts

```diff
@@ -0,0 +1,216 @@
+    selected: boolean,
+    valueCode: string,
+    valueDisplay: string
+  ): Configurator.Value => {
+    const value: Configurator.Value = {
+      description,
+      images,
+      quantity,
+      selected,
+      valueCode,
+      valueDisplay,
+    };
+    return value;
+  };
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [
+          I18nTestingModule,
+          ReactiveFormsModule,
+          RouterTestingModule,
+          UrlTestingModule,
+        ],
+        declarations: [
+          ConfiguratorAttributeProductCardComponent,
+          ConfiguratorShowMoreComponent,
+          ItemCounterComponent,
+        ],
+      })
+        .overrideComponent(ConfiguratorAttributeProductCardComponent, {
+          set: {
+            changeDetection: ChangeDetectionStrategy.Default,
+          },
+        })
+        .compileComponents();
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      ConfiguratorAttributeProductCardComponent
+    );
+    component = fixture.componentInstance;
+    component.quantity = mockQuantity;
+    component.disabledAction = false;
+    component.multiSelect = false;
+    component.product = createValue(
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+      [createImage('url', 'alt')],
+      1,
+      false,
+      '1111-2222',
+      'Lorem Ipsum Dolor'
+    );
+
+    spyOn(component, 'onHandleSelect').and.callThrough();
+    spyOn(component, 'onHandleDeselect').and.callThrough();
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should button be enabled when disabledAction = false & selected = false', () => {
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+    expect(button.disabled).toBe(false);
+  });
+
+  it('should button be enabled when disabledAction = false & selected = true', () => {
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+    expect(button.disabled).toBe(false);
+  });
+
+  it('should button be disabled when disabledAction = true & selected = true', () => {
+    component.disabledAction = true;
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+
+    expect(button.disabled).toBe(true);
+  });
+
+  it('should button be called with onHandleSelect', () => {
```

Comment:
use more generic name for the test, something like `... be called with proper method` or something similar

Reply:
renamed